express-notes-api/index.js

```diff
@@ -0,0 +1,150 @@
+const express = require('express');
+const app = express();
+const data = require('./data.json');
+const JSONMiddleware = express.json();
+app.use(JSONMiddleware);
+
+// require automatically parses
+
+app.get('/api/notes', function (req, res) {
+  const array = [];
+  for (const property in data.notes) {
+    array.push(data.notes[property]);
+  }
+  res.status(200);
+  res.send(array);
+});
+
+app.get('/api/notes/:id', function (req, res) {
+  const reqId = Number(req.params.id);
+  if (isNaN(reqId)) {
+    const errorObject = {
+      error: 'id must be a positive integer'
+    };
+    res.send(errorObject);
+    res.sendStatus(400);
```

Comment:
You'll want to use `status()` (instead of `sendStatus()`) _before_ `send()`. Double-check your status codes with HTTPie.